File: docs/versioned_docs/version-1.12.0/API-Reference/javascript-examples/api-projects/import-a-project.js
const fs = require("fs");
const path = require("path");

const fixturesDir = path.join(__dirname, "../../fixtures");
const defaultProjectZip = path.join(fixturesDir, "project-import.zip");
const projectPath = process.env.PROJECT_IMPORT_FILE || defaultProjectZip;
const projectBuf = fs.readFileSync(projectPath);
const projectName = path.basename(projectPath);

const url = `${process.env.LANGFLOW_URL ?? ""}/api/v1/projects/upload/`;

const formData = new FormData();
formData.append("file", new Blob([projectBuf], { type: "application/zip" }), projectName);

const options = {
  method: "POST",
  headers: {
    accept: "application/json",
    "x-api-key": `${process.env.LANGFLOW_API_KEY ?? ""}`,
  },
  body: formData,
};

fetch(url, options)
  .then(async (response) => {
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    const text = await response.text();
    console.log(text);
  })
  .catch((error) => console.error(error));

File: docs/versioned_docs/version-1.12.0/API-Reference/python-examples/api-projects/import-a-project.py
import os
from pathlib import Path

import requests

base = os.environ.get("LANGFLOW_URL", "")
api_key = os.environ.get("LANGFLOW_API_KEY", "")

fixtures = Path(__file__).resolve().parents[2] / "fixtures"
default_json = fixtures / "project-import.json"
import_path = Path(os.environ.get("PROJECT_IMPORT_JSON", str(default_json)))

headers = {"accept": "application/json", "x-api-key": api_key}

files = {"file": (import_path.name, import_path.read_bytes(), "application/json")}
response = requests.post(f"{base}/api/v1/projects/upload/", headers=headers, files=files, timeout=60)

response.raise_for_status()
print(response.text)
